packages/belos/src/BelosPseudoBlockCGIter.hpp

```diff
@@ -505,25 +506,25 @@ namespace Belos {
       //
       // Update the search directions.
       for (i=0; i<numRHS_; ++i) {
-        beta(i,i) = rHz[i] / rHz_old[i];
+        beta[i] = rHz[i] / rHz_old[i];
         index[0] = i;
         Teuchos::RCP<const MV> Z_i = MVT::CloneView( *Z_, index );
         Teuchos::RCP<MV> P_i = MVT::CloneViewNonConst( *P_, index );
-        MVT::MvAddMv( one, *Z_i, beta(i,i), *P_i, *P_i );       
+        MVT::MvAddMv( one, *Z_i, beta[i], *P_i, *P_i );       
       }
 
       // Condition estimate (if needed)
-      if(doCondEst_ > 0) {
-        if(iter_ > 1 ) {
-          diag_[iter_-1]    = Teuchos::ScalarTraits<ScalarType>::real((beta_old * beta_old * pAp_old + pAp[0]) / rHz_old[0]);
-          offdiag_[iter_-2] = -Teuchos::ScalarTraits<ScalarType>::real(beta_old * pAp_old / (sqrt( rHz_old[0] * rHz_old2)));
+      if (doCondEst_) {
+        if (iter_ > 1) {
+          diag_[iter_-1]    = Teuchos::ScalarTraits<ScalarType>::real((beta_old_ * beta_old_ * pAp_old_ + pAp[0]) / rHz_old[0]);
+          offdiag_[iter_-2] = -Teuchos::ScalarTraits<ScalarType>::real(beta_old_ * pAp_old_ / (sqrt( rHz_old[0] * rHz_old2_)));
```

Comment:
See above comment (`rhs_old` is an array of `ScalarType`, and you're taking its `sqrt` here).  You may need to `real` it first etc.